course/page/base.py

```diff
@@ -388,8 +389,9 @@ def __init__(self, vctx, location, page_desc):
             self.is_optional_page = getattr(page_desc, "is_optional_page", False)
 
         else:
-            from warnings import warn
-            warn(_("Not passing page_desc to PageBase.__init__ is deprecated"),
+            if vctx is not None:
+                vctx.add_warning(
+                    _("Not passing page_desc to PageBase.__init__ is deprecated"),
```

Comment:
This doesn't have much to do with course content--I don't think this warning belongs in validation output.

Reply:
You're right, Actually I'm now reverting that.